### PR TITLE
Googlemaps api

### PR DESCRIPTION
**Pull request by clemcodes**

### CODE REVIEW

**Review by Sravya12379 — APPROVED** [review]

File: googlemaps.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import requests
+
+
+def input_locations():
+    """
+    This allows the user to input the locations of the origin and destination and then puts these into the function to
+    get the distance
```
Comment (robynfsj) [inline googlemaps.py:7]: Love the function documentations! I think that in production code the descriptions might be a bit more accurate. These are probably more helpful to us at the moment but we might want to change them before we submit. Technically this function doesn't put the origin and destination strings into the the get_distance function. It just returns origin and destination. A more accurate description would be:

This allows the user to input the locations of the origin and destination, assigns them to variables and returns these variables.

File: googlemaps.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    return origin, destination
+
+
+def get_distance(origin, destination):
+    """
+    This function takes the inputted origin and destination, then uses the google maps API to calculate the distances
+    and duration (take out if not needed) of the route. These are outputted for each mode of transport so they are all
+    available.
+
+    :param origin: inputted origin
```
Comment (robynfsj) [inline googlemaps.py:22]: You can also add the data type if you wanted to make it more clear. The data type will then appear next to the parameter when you hover over the function name in the IDE. e.g.

`:type origin: string`

Example from my hw this week:
Screenshot 2021-08-06 at 10 01 04

File: googlemaps.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                distance = data['distance']['text']
+                duration = data['duration']['text']
+
+        distances[mode] = distance
+        durations[mode] = duration
+
+    return origin_address, destination_address, distances, durations
+
+
+origin, destination = input_locations()
```
Comment (robynfsj) [inline googlemaps.py:54]: Just a note to say we will want to remove this example run through from the script later.

File: googlemaps.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    api_key = "Your API keys"
+    modes = ["driving", "walking", "bicycling", "transit"]
+    distances = dict()
+    durations = dict()
+    for mode in modes:
+        uri = f'https://maps.googleapis.com/maps/api/distancematrix/json?' \
+              f'origins={origin}&' \
+              f'destinations={destination}&' \
+              f'mode={mode}&' \
+              f'key={api_key}&language=en-GB'
```
Comment (robynfsj) [inline googlemaps.py:35]: Maybe add a comment somewhere to remind person running script to add their API key in here.

**Review by robynfsj — APPROVED** [review]

File: googlemaps.py (hunk 1 above)
Comment (clemcodes) [reply googlemaps.py:7]: Cheers, updated!

File: googlemaps.py (hunk 2 above)
Comment (clemcodes) [reply googlemaps.py:22]: Nice to know that, learned a lot :D updated!

File: googlemaps.py (hunk 4 above)
Comment (clemcodes) [reply googlemaps.py:35]: API keys are supposed to be added at line 25 - ` api_key = "Your API keys"`
I updated it to ` api_key = "Your API key here"` on the merged main branch on line 28 and hopefully it's clearer now

**Pull request merged**